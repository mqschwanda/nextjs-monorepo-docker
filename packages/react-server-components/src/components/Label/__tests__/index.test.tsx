import { render } from '@testing-library/react';
import { Label } from '..';

describe('@mqs/react-server-components', () => {
  describe('components', () => {
    describe('<Label />', () => {
      it('renders', () => {
        const testId = Label.name;

        const { getByTestId, asFragment } = render((
          <Label
            className='test'
            testId={testId}
          >
            <span
              className='label-text'
            >
              { 'What is your name?' }
            </span>
            <span
              className='label-text-alt'
            >
              { 'Top Right label' }
            </span>
          </Label>
        ));

        const element = getByTestId(testId);
        expect(element).toBeTruthy();

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });
});
