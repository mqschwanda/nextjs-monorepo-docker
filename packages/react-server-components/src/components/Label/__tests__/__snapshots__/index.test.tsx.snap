// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@mqs/react-server-components components <Label /> renders 1`] = `
<DocumentFragment>
  <label
    class="label test"
    data-testid="Label"
  >
    <span
      class="label-text"
    >
      What is your name?
    </span>
    <span
      class="label-text-alt"
    >
      Top Right label
    </span>
  </label>
</DocumentFragment>
`;
